src/lambdaisland/janus.clj

```diff
@@ -0,0 +1,87 @@
+(ns lambdaisland.janus
+  (:require [clojure.string :as str])
+  (:import [com.vladsch.flexmark.ast Heading BulletList BulletListItem]
+           [com.vladsch.flexmark.parser Parser]
+           [com.vladsch.flexmark.util.ast Block Node]))
+
+(set! *warn-on-reflection* true)
+
+(defn remove-nil-vals [item]
+  (into {} (filter (comp some? val)) item))
+
+(defn node-text
+  [^Block node]
+  (when node
+    (.toString (.getChars node))))
+
+(defn extract-version-components
+  "Extracts version-id, date and sha data from a given string
+  e.g. \"0.0-601 (2020-03-11 / 6b88d96)\""
+  [s]
+  (when-let [[_ version date sha] (re-find #"#\s+([^ ]+)(?:\s+\(([0-9-]+)?(?:[ /]+([a-f0-9]+)?)?\))?" s)]
```

Comment:
I made this a bit more forgiving so it will still work if there is only a version without a date or sha